### PR TITLE
(FB: 154747) - Implementation of UV Submit an Idea

### PR DESCRIPTION
**Pull request by benrudolph**
Allows users to submit ideas from CommCareHQ

This commit takes care of both these tickets:
http://manage.dimagi.com/default.asp?154747
http://manage.dimagi.com/default.asp?155764

### CODE REVIEW

File: corehq/apps/style/static/style/js/uv.widget.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        uvBtnId = '#uvSubmitIdea',
+        $uvBtn = $(uvBtnId);
+
+    // Return if the submit an idea button doesn't exist
+    if (!$uvBtn.length) {
+        return;
+    }
+
+    // Include the UserVoice JavaScript SDK (only needed once on a page)
+    window.UserVoice=window.UserVoice||[];(function(){var uv=document.createElement('script');uv.type='text/javascript';uv.async=true;uv.src='//widget.uservoice.com/PeUfQwynga75cBUaAV8Ew.js';var s=document.getElementsByTagName('script')[0];s.parentNode.insertBefore(uv,s)})();
```
Comment (dimagimon) [inline corehq/apps/style/static/style/js/uv.widget.js]: Missing semicolon.

File: corehq/apps/style/static/style/js/uv.widget.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    //
+    // UserVoice Javascript SDK developer documentation:
+    // https://www.uservoice.com/o/javascript-sdk
+    //
+
+    // UV crashes hard when it receives undefined as a value. Need to ensure defaults.
+    defaults = {
+        userEmail: 'Unknown email address',
+        userName: 'Unknown name'
+    }
```
Comment (dimagimon) [inline corehq/apps/style/static/style/js/uv.widget.js]: Missing semicolon.

File: corehq/apps/style/static/style/js/uv.widget.js (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/style/static/style/js/uv.widget.js]: This is copied and pasted from UV website

**Comment by benrudolph** [comment]
Here are screenshots of what the widget looks like:

screen shot 2015-02-10 at 1 55 53 pm
screen shot 2015-02-10 at 1 56 13 pm

**Comment by snopoke** [comment]
Would it be possible to get it to auto-populate the email field with the current user's email?

**Comment by benrudolph** [comment]
As far as I know, no. We do send the email to UV via `identify`. Let me read the docs some more and talk to UV people as some of the functionality isn't documented.

**Pull request merged**